File: app/src/main/java/com/faendir/lightning_launcher/multitool/util/notification/NotificationListener.java
package com.faendir.lightning_launcher.multitool.util.notification;

import android.service.notification.NotificationListenerService;
import android.service.notification.StatusBarNotification;

public interface NotificationListener {

    default void onCreate(NotificationListenerService context) {
    }

    default void onNotificationPosted(NotificationListenerService context, StatusBarNotification sbn) {
    }

    default void onNotificationRemoved(NotificationListenerService context, StatusBarNotification sbn) {
    }
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/util/notification/NotificationListener.kt
package com.faendir.lightning_launcher.multitool.util.notification

import android.service.notification.NotificationListenerService
import android.service.notification.StatusBarNotification

interface NotificationListener {

    fun onCreate(context: NotificationListenerService) {}

    fun onNotificationPosted(context: NotificationListenerService, sbn: StatusBarNotification) {}

    fun onNotificationRemoved(context: NotificationListenerService, sbn: StatusBarNotification) {}
}
